### PR TITLE
Fixes for building

### PR DESCRIPTION
**Pull request by JoshuaWatt**
Fixes some issues found while trying to build wpewebkit

### CODE REVIEW

**Comment by philn** [comment]
@JoshuaWatt thanks for the PR, I'll try to test it soon. We don't use change-ids in this project BTW, so feel free to remove them from the commit messages :)

**Comment by JoshuaWatt** [comment]
Oops. Sorry about that. I have a global git hook to install them for our use in Gerrit, and I sometimes forget to disable it on new repos. I'll remove them

File: recipes-browser/wpewebkit/wpewebkit.inc [hunk 1]
```diff
@@ -12,7 +12,8 @@ DEPENDS = " \
     libsoup-2.4 libwebp libxml2 libxslt virtual/egl virtual/libgles2 libepoxy libgcrypt \
 "
 
-inherit cmake pkgconfig perlnative pythonnative cmake_qt5
+inherit cmake pkgconfig perlnative pythonnative
+inherit ${@'cmake_qt5' if 'qt5-layer' in d.getVar('BBFILE_COLLECTIONS').split() else ''}
```
Comment (philn) [inline recipes-browser/wpewebkit/wpewebkit.inc]: Should the qtwpe PACKAGECONFIG option remain as-is? It tracks some qt deps recipes.

Comment (JoshuaWatt) [reply recipes-browser/wpewebkit/wpewebkit.inc]: I think it's OK as is. The default PACAKGECONFIG for the recipe doesn't include qtwpe, so it won't cause failures if you don't have meta-qt5. If override the default PACKAGECONFIG to add qtwpe but don't add the meta-qt5 layer, you'll get errors pretty quickly for the missing qt build time dependencies.

**Comment by philn** [comment]
I tried this but it fails:
```
ERROR: Nothing RPROVIDES 'qt-wpe-simple-browser' (but .../images/core-image-weston-wpe.bb RDEPENDS on or otherwise requires it)
NOTE: Runtime target 'qt-wpe-simple-browser' is unbuildable, removing...
Missing or unbuildable dependency chain was: ['qt-wpe-simple-browser'] 
ERROR: Required build target 'core-image-weston-wpe' has no buildable providers.
Missing or unbuildable dependency chain was: ['core-image-weston-wpe', 'qt-wpe-simple-browser']
```

**Comment by JoshuaWatt** [comment]
Ah, I messed the image recipe. I'll update it

**Comment by JoshuaWatt** [comment]
Ok, I had used the wrong name in the dynamic layer path. It should be fixed now

**Pull request merged**